sfge/src/main/java/com/salesforce/graph/vertex/DmlStatementVertex.java

```diff
@@ -1,9 +1,94 @@
 package com.salesforce.graph.vertex;
 
+import com.salesforce.apex.jorje.ASTConstants;
+import java.util.List;
 import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
+/** Represents a vertex that can perform DML through Apex. */
 public abstract class DmlStatementVertex extends BaseSFVertex {
+    protected static final Logger LOGGER = LogManager.getLogger(DmlStatementVertex.class);
+
+    private static final String ACCESS_LEVEL_REFERENCE = "AccessLevel";
+
+    private enum AccessLevel {
+        USER_MODE,
+        SYSTEM_MODE
+    }
+
+    private final AccessLevel accessLevel;
+
     DmlStatementVertex(Map<Object, Object> properties) {
         super(properties);
+        accessLevel = calculateAccessLevel(this);
+    }
+
+    private static AccessLevel calculateAccessLevel(DmlStatementVertex vertex) {
+        // Default to System mode to begin with.
+        AccessLevel accessLevel = AccessLevel.SYSTEM_MODE;
+
+        // If AccessLevel is included in the syntax, it's usually the last child. It shows up in the
```

Comment:
"Usually the last child" are there known cases where it won't be? Will those cases break here?

Reply:
When "as user" or "as system" is not included, the variable that contains the value for DML is the last child. If "as user" or "as system" is included, it'll be the last child. The conditionals below confirm other things in the code that follows.